### ISSUE TITLE
Economy Improvements

### ISSUE THREAD

**Issue by bobbens** [naev/naev#1008]
We have a basic economy, however, there is still lots of room for improvements. Some ideas that were thrown around:
* Lua hooks to allow events/missions to manipulate economy (and consequently implement some events)
* Possibly also move most of the economy stuff to Lua if it is fast enough.
* Replace the sinusoidal stuff by a wiener process (@nenau-again)
* Add some way not to have to hardcode the characteristics of the planets and of the commodities (@nenau-again)
* Replace the averaging process by a sparse linalg operation (with impedance) @bobbens told me that he was not satisfied with his own attempt for that, but I don't know if the issue is that he has a problem to solve the linalg problem, or if the process works well, but the result is not good looking. (@nenau-again)

**Comment by nenau-again** [comment]
A few remarks about my points : 
* "Add some way not to have to hardcode [...]" what I was talking about here has been done as data has been switched to xml.
* "Replace the averaging process by a sparse linalg operation [...]" : Maybe it could be possible to use an explicit algorithm instead (with a few tricks to ensure numerical stability). This would make the computational cost negligible.

**Comment by nenau-again** [comment]
Yes, my thoughts are not well-ordered. Here is something hopefully clearer :

In your short-range averaging process, no local event (a factory stops producing something, or someone buys lots of one commodity) can change the economical state far away.
In our case, it's not necessary a problem since we don't want to model a real economy, but maybe at some point we will find interesting to model long-range interactions (mainly for rare commodities). For that, afaik, we could use the heat equation (with a forcing term that follows a wiener process I guess). https://en.wikipedia.org/wiki/Heat_equation
Or alternatively Laplacian equation, which is it's static counterpart (time is not taken into account).

Once the equation is space-discretized, the standard, implicit algorithm, requires one matrix inversion (linalg operation) at each time-step (ie any landing). This scheme can be made unconditionally stable (ie never diverges to stupid values provided one is careful enough), but requires linalg operations, that could be quite CPU-costly.
An other algorithm, the explicit one, could be carried on without any matrix inversion, and requires only averaging operations at each time step (ie alnding). However, this algo can be unstable if the time step is too large (ie may diverge to stupid values) unless one does cowboyish tricks.

Anyway, as you see, it's complex stuff to solve an hypethetic issue that could probably very well stay unresolved. If what I wrote sounds alien to you, I guess I could give a try at some point in the far future after I have done the tons of real-life work, and the tons of naev work that I still have pending.

**Comment by agb32** [comment]
Latest pull request #1080 adds economy information onto the map, so that you can tell at a glance whether a trade with a certain destination will be good or bad.
